src/messaging/rabbitmq/consumer.ts

```diff
@@ -90,15 +96,58 @@ export default class RMQConsumer {
     })
   }
 
-  public async cleanUp(): Promise<void> {
-    this.isConnClosing = true
-    if (this.channel != null) {
-      await this.channel.close()
-      console.log(`[Consumer ${this.name}]: Closed channel successfully`)
-    }
-    if (this.conn !== null) {
-      await this.conn.close()
-      console.log(`[Consumer ${this.name}]: Closed connection successfully`)
+  public async cleanUp(isConnClosing = false): Promise<void> {
+    this.isConnClosing = isConnClosing
+
+    try {
+      // Check if channel exists and is still open
+      if (this.channel?.connection) {
+        try {
+          console.log(`[Consumer ${this.name}]: Nacking all...`)
+          await Promise.race([
+            this.channel.nackAll(true),
+            new Promise((_, reject) => setTimeout(() => reject(new Error('nackAll timeout')), 10000)),
+          ])
+        } catch (error) {
+          console.warn(`[Consumer ${this.name}]: Error or timeout in nackAll - ${error}`)
+        }
+
+        try {
+          console.log(`[Consumer ${this.name}]: Closing channel...`)
+          await Promise.race([
+            this.channel.close(),
```

Comment:
Improved this handling a bit by adding timeout of 10s. In reconnect cases, earlier implementation would stuck. 

In reconnect flow, this doesn't work because channels are already closed. This could be because the scenario we are trying to test reconnect flow. In some other scenarios, channel closing would work(like exit listeners or unexpected application errors).
This is a good handling to have in general to avoid connection leaks.